app/src/main/java/by/naxa/soundrecorder/fragments/RecordFragment.java

```diff
@@ -236,6 +238,9 @@ public void onSingleClick(View v) {
                     if (PermissionsHelper.checkAndRequestPermissions(
                             RecordFragment.this, MY_PERMISSIONS_REQUEST_RECORD_AUDIO)) {
                         startRecording();
+                    } else {
+                        Toast.makeText(getActivity(), getString(R.string.error_no_permission_granted_record), Toast.LENGTH_LONG).show();
+                        openAppSettings();
```

Comment:
This is a handler for the RECORD button. I assume that the PAUSE/RESUME button is affected by the same issue, so it would be good if you apply your fix in both places. Though I didn't test permissions request after click on RESUME button. So not sure. But I suggest you to make changes in `PermissionsHelper.checkAndRequestPermissions` method, which is invoked after a click on RECORD button as well as a click on RESUME button.